visual-dotnet/SauceLabs.Visual/BuildFactory.cs

```diff
@@ -90,20 +90,13 @@ internal static async Task CloseBuilds()
         /// <exception cref="VisualClientException">when build is not existing or has an invalid state</exception>
         private static async Task<VisualBuild> FindBuildById(VisualApi api, string buildId)
         {
-            try
-            {
-                var build = (await api.Build(buildId)).EnsureValidResponse().Result;
-                if (build == null)
-                {
-                    throw new VisualClientException($@"build {buildId} was not found");
-                }
-
-                return new VisualBuild(build.Id, build.Url, build.Mode);
-            }
-            catch (VisualClientException)
+            var build = (await api.Build(buildId)).EnsureValidResponse().Result;
```

Comment:
We still need to have a try/catch block here because GraphQLResponseExtension in the EnsureValidResponse method throws it - lack of the `buildId` can be problematic there.